Classes/Emogrifier.php

```diff
@@ -681,8 +692,22 @@ private function getUnifiedHtml()
         } else {
             $bodyWithoutUnprocessableTags = $this->html;
         }
+        
+        if ($this->shouldFixMissingCharsetDeclaration) {
```

Comment:
Please lets move this block (of the block in the inner or outer if) into a well-named, private method.